arch/arm/src/gd32f4/gd32f4xx_serial.c

```diff
@@ -159,6 +158,7 @@ struct up_dev_s
   const uint8_t   stop_2bits;    /* True: Configure with 2 stop bits instead of 1 */
   const uint32_t  tx_gpio;       /* USART TX GPIO pin configuration */
   const uint32_t  rx_gpio;       /* USART RX GPIO pin configuration */
+  spinlock_t      lock;          /* Spinlock */
```

Comment:
Is this change intended to be included in same PR?

Reply:
yes， It also resulted in compilation errors.
Configuration/Tool: gd32f470zk-eval/nsh,CONFIG_ARM_TOOLCHAIN_GNU_EABI
2024-12-27 02:06:08
------------------------------------------------------------------------------------
  Cleaning...
  Configuring...
  Disabling CONFIG_ARM_TOOLCHAIN_GNU_EABI
  Enabling CONFIG_ARM_TOOLCHAIN_GNU_EABI
  Building NuttX...
Error: chip/gd32f4xx_serial.c:441:4: error: 'struct up_dev_s' has no member named 'lock'; did you mean 'clock'?
  441 |   .lock          = SP_UNLOCKED,
      |    ^~~~
      |    clock
In file included from chip/gd32f4xx_serial.c:42:
chip/gd32f4xx_serial.c: In function 'up_disableusartint':
Error: chip/gd32f4xx_serial.c:1139:36: error: 'struct up_dev_s' has no member named 'lock'; did you mean 'clock'?
 1139 |   flags = spin_lock_irqsave(&priv->lock);
      |                                    ^~~~
Error: chip/gd32f4xx_serial.c:1173:33: error: 'struct up_dev_s' has no member named 'lock'; did you mean 'clock'?
 1173 |   spin_unlock_irqrestore(&priv->lock, flags);
      |                                 ^~~~
chip/gd32f4xx_serial.c: In function 'up_restoreusartint':
Error: chip/gd32f4xx_serial.c:1184:36: error: 'struct up_dev_s' has no member named 'lock'; did you mean 'clock'?
 1184 |   flags = spin_lock_irqsave(&priv->lock);
      |                                    ^~~~
Error: chip/gd32f4xx_serial.c:1188:33: error: 'struct up_dev_s' has no member named 'lock'; did you mean 'clock'?
 1188 |   spin_unlock_irqrestore(&priv->lock, flags);
      |                                 ^~~~